README.md

```diff
@@ -20,3 +20,98 @@ cli-kit new myproject
 
 You can see example usage [here](https://github.com/Shopify/cli-kit-example). This app is similar to
 the one that the generator generates.
+
+## Starting a New Project
+To begin creating your first `cli-kit` application, run:
+```bash
+cli-kit new myproject
+```
+Where `myproject` is the name of the application you wish to create.  Then, you will be prompted to
+select how the project consumes `cli-kit` and `cli-ui`.  The available options are:
+- Vendor (faster execution, more difficult to update dependencies)
+- Bundler (slower execution, easier dependency management)
+
+You're now ready to write your very first `cli-kit` application!
+
+## How do `cli-kit` Applications Work?
+The executable for your `cli-kit` app is stored in the "exe" directory.  To execute the app, simply
+run:
+```bash
+./exe/myproject
+```
+
+### Folder Structure
+* `/exe/` - Location of the executables for your application.
+* `/lib/` - Location of the resources for your app (modules, classes, helpers, etc).
+    * `myproject.rb` - The main file for your application.
+    * `myproject/` - Stores the various commands/entry points.
+        * `entry_point.rb` - Is executed when the application is launched, and handles commands.
+        * `commands.rb` - Registers the various commands that your application is able to handle.
+        * `commands/` - Stores Ruby files for each command (help, new, add, etc).
+
+## Adding a New Command to your App
+### Registering the Command
+Let's say that you'd like your program to be able to handle a specific task, and you'd like to
+_register_ a new handler for the command for that task, like `todo add` to add an item to a to-do
+list.
+To do this, open `/lib/myproject/commands.rb`, where `myproject` is the name of your app.  Then, add
```

Comment:
~where `myproject` is the name of your app~ you established that above, and it's fairly clear.  Repeating it just interrupts the flow of what you're saying.